packages/integration-tests/index.js

```diff
@@ -502,24 +506,49 @@ async function cleanupProviders(stackName, bucket, providersDirectory, postfix)
  *
  * @param {string} config - Test config used to set environment variables and template rules data
  * @param {string} dataDirectory - the directory of rules json files
- * @param {string} overrides - override rule fields
+ * @param {Object} overrides - override rule fields
+ * @param {string} postfix - string to append to rule name, collection, and provider
  * @returns {Promise.<Array>} array of Rules added
  */
-async function addRules(config, dataDirectory, overrides) {
+async function addRulesWithPostfix(config, dataDirectory, overrides, postfix) {
   const { stackName, bucket } = config;
   const rules = await setupSeedData(stackName, bucket, dataDirectory);
 
-  const promises = rules.map((rule) => limit(() => {
+  // Rules should be added in serial because in the case of SNS and Kinesis rule types
+  // they may share an event source mapping and running them in parallel will cause a
+  // race condition
+  const ruleLimit = pLimit(1);
+
+  const promises = rules.map((rule) => ruleLimit(() => {
+    if (postfix) {
+      rule.name += globalReplace(postfix, '-', '_'); // rule cannot have dashes
```

Comment:
This is mutating the rule object that is passed in.  That kind of thing has bitten us before.  Better to create a copy of the rule and modify _that_.

Reply:
I take this back, rules are not arguments.